src/main/java/com/powsybl/openloadflow/ac/outerloop/SecondaryVoltageControlOuterLoop.java

```diff
@@ -248,37 +274,55 @@ private static SensiVq calculateSensiVq(SensitivityContext sensitivityContext, L
      * dvi = dq / sqi
      * </pre>
      */
-    private static void adjustPrimaryVoltageControlTargets(String zoneName, SensitivityContext sensitivityContext,
-                                                           List<LfBus> controlledBuses, LfBus pilotBus, double pilotDv) {
+    private boolean adjustPrimaryVoltageControlTargets(String zoneName, SensitivityContext sensitivityContext,
+                                                       List<LfBus> controlledBuses, LfBus pilotBus, double pilotDv) {
+        boolean adjusted = false;
+
         // calculate sensitivity of controlled buses voltage to pilot bus voltage
         double[] svi = sensitivityContext.calculateSensiVv(controlledBuses, pilotBus);
 
         // calculate sensitivity of controlled buses voltage to controller buses reactive power injection
         // and sensitivity of controller bus reactive power injection to pilot bus voltage
-        var sensiVq = calculateSensiVq(sensitivityContext, controlledBuses, svi);
-
-        // supposing we want all controllers to shift to same amount of reactive power
-        double dq = pilotDv / Arrays.stream(sensiVq.si).sum();
-
-        LOGGER.trace("Control units of zone '{}' need to be adjusted of {} MVar", zoneName, dq * PerUnit.SB);
-
-        for (LfBus controlledBus : controlledBuses) {
-            double pvcDv = 0;
-            for (LfBus controllerBus : getControllerBuses(controlledBus)) {
-                double sqi = sensiVq.getSqi(controllerBus);
-                if (sqi != 0) {
-                    pvcDv += dq / sqi;
+        var sensiVq = calculateSensiVq(zoneName, sensitivityContext, controlledBuses, svi);
+
+        double siSum = Arrays.stream(sensiVq.si).sum();
+        if (siSum != 0) {
+            // supposing we want all controllers to shift to same amount of reactive power
+            double dq = pilotDv / siSum;
+            if (Math.abs(dq) > DQ_PU_EPS) {
+                LOGGER.trace("Each of the controller buses of zone '{}' need to be adjusted of {} MVar", zoneName, dq * PerUnit.SB);
+
+                for (LfBus controlledBus : controlledBuses) {
+                    double pvcDv = 0;
+                    for (LfBus controllerBus : getControllerBuses(controlledBus)) {
+                        double sqi = sensiVq.getSqi(controllerBus);
+                        if (sqi != 0) {
+                            pvcDv += dq / sqi;
+                        }
+                    }
+                    if (pvcDv != 0) {
+                        var pvc = controlledBus.getGeneratorVoltageControl().orElseThrow();
+                        double newPvcTargetV = pvc.getTargetValue() + pvcDv;
+                        String plausibleTargetInfos = "";
+                        if (newPvcTargetV > maxPlausibleTargetVoltage) {
```

Comment:
You are "cutting" the targets. It was what we have decided to do for the load flow at beginning bu we have changed to discarded the voltage control. I imagine that we cannot discarded the voltage control here. We will have an incoherence, is it a real issue or not?